src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,194 @@
+     *
+     * @param deadlineDescription  The input command to parse.
+     * @param tasks   The list of tasks.
+     * @param ui      The UI for user interactions.
+     * @param storage The storage for tasks.
+     * @return The response message after adding the task.
+     * @throws DukeException If the command is invalid or execution fails.
+     */
+    private static String addDeadline(String input, TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        String[] splitInput = input.split(" /by ");
+        if (splitInput.length < 2 || splitInput[0].isEmpty() || splitInput[1].isEmpty()) {
+            return "The deadline description or date is missing.";
+        }
+        Deadline newDeadline = new Deadline(splitInput[0].trim(), splitInput[1].trim());
+        if (!tasks.isDuplicate(newDeadline)) {
+            tasks.addTask(newDeadline);
+            storage.save(tasks.getTasks());
+            return ui.showTaskAdded(newDeadline, tasks.getSize());
+        } else {
+            return "This deadline is a duplicate and won't be added.";
+        }
+    }
+
+
+
+    /**
+     * Adds a event task to the list of tasks.
+     *
+     * @param eventDescription  The input command to parse.
+     * @param tasks   The list of tasks.
+     * @param ui      The UI for user interactions.
+     * @param storage The storage for tasks.
+     * @return The response message after adding the task.
+     * @throws DukeException If the command is invalid or execution fails.
+     */
+    private static String addEvent(String input, TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        String[] splitInput = input.split(" /from ");
+        if (splitInput.length < 2 || splitInput[0].isEmpty() || splitInput[1].isEmpty()) {
+            return "The event description or start time is missing.";
+        }
+        String[] timeSplit = splitInput[1].split(" /to ");
+        if (timeSplit.length < 2 || timeSplit[0].isEmpty() || timeSplit[1].isEmpty()) {
+            return "The event end time or details are missing.";
+        }
+        Event newEvent = new Event(splitInput[0].trim(), timeSplit[0].trim(), timeSplit[1].trim());
+        if (!tasks.isDuplicate(newEvent)) {
+            tasks.addTask(newEvent);
+            storage.save(tasks.getTasks());
+            return ui.showTaskAdded(newEvent, tasks.getSize());
+        } else {
+            return "This event is a duplicate and won't be added.";
+        }
+    }
+
+    /**
+     * Marks or unmarks a task in the list of tasks.
+     *
+     * @param description  The input command to parse.
+     * @param tasks   The list of tasks.
+     * @param ui      The UI for user interactions.
+     * @param storage The storage for tasks.
+     * @return The response message after marking or unmarking the task.
+     * @throws DukeException If the command is invalid or execution fails.
+     */
+    private static String markOrUnmarkTask(String description, TaskList tasks, Ui ui,
```

Comment:
I am unsure of this convention. Is this the convention we are supposed to follow? Can help me anyone?